README.markdown

```diff
@@ -22,3 +22,51 @@ long the outcome is distributed under the same license. Otherwise, you must
 obtain a [commercial license](./LICENSE-COMMERCIAL) that removes such
 restrictions. Read more about our licensing approach
 [here](https://www.sourcemeta.com/licensing/).
+
+Example
+--------------
+
+```cpp
+#include <sourcemeta/jsontoolkit/json.h>
+#include <sourcemeta/jsontoolkit/jsonpointer.h>
+#include <sstream>
+#include <iostream>
+
+int main()
+{
+    // creating json using parse
+    sourcemeta::jsontoolkit::JSON _template_ =
+        sourcemeta::jsontoolkit::parse(R"JSON({
+            "name" : "xyz" ,
+            "age" : 20 ,
+            "address" : "zxy"
+        })JSON");
+
+    // stringstream for output
+    std::ostringstream stringvar;
+    // pointer for the name
+    const sourcemeta::jsontoolkit::Pointer name_pointer{"name"};
+    const sourcemeta::jsontoolkit::Pointer age_pointer{"age"};
+    const sourcemeta::jsontoolkit::Pointer address_pointer{"address"};
+    // value instance for json
+    const sourcemeta::jsontoolkit::JSON name_value{"xyz_with_foo"};
+    const sourcemeta::jsontoolkit::JSON age_value{20 + 1};
+    const sourcemeta::jsontoolkit::JSON address_value{"zxy_with_bar"};
+    // updating the value of the name using pointer
+    sourcemeta::jsontoolkit::set(_template_, name_pointer, name_value);
+    sourcemeta::jsontoolkit::set(_template_, age_pointer, age_value);
+    sourcemeta::jsontoolkit::set(_template_, address_pointer, address_value);
+    sourcemeta::jsontoolkit::prettify(_template_, stringvar);
+    std::cout << stringvar.str() << std::endl;
+
+    /*
```

Comment:
To style this better, you can say something like:

```
// The above program will print the following to standard output:
//
// {
//   ...
// }
```

Reply:
And please try to follow the style of the project for examples. Nowhere in the codebase we use `/* */` comment styles, but `//` (and `///` for Doxygen). It is important to be consistent!